File: spendy_backend/src/main/java/com/rohitcodes/expense_tracker/service/BudgetService.java
package com.rohitcodes.expense_tracker.service;

import com.rohitcodes.expense_tracker.dto.budget.BudgetRequest;
import com.rohitcodes.expense_tracker.dto.budget.BudgetResponse;

import java.util.List;

public interface BudgetService {

    BudgetResponse createBudget(Long userId, BudgetRequest request);

    List<BudgetResponse> getBudgets(Long userId, Integer budgetYear, Integer budgetMonth);

    BudgetResponse getBudgetById(Long userId, Long budgetId);

    BudgetResponse updateBudget(Long userId, Long budgetId, BudgetRequest request);

    void deleteBudget(Long userId, Long budgetId);
}

File: spendy-mobileFrontend/src/services/budgetService.ts
import { api } from "@/src/services/api";
import type { Budget, BudgetPayload } from "@/src/types/budget";

export const budgetService = {
  async getBudgets(userId: number, params?: { budgetYear?: number; budgetMonth?: number }) {
    const response = await api.get<Budget[]>(`/users/${userId}/budgets`, {
      params
    });
    return response.data;
  },

  async getBudgetById(userId: number, budgetId: number) {
    const response = await api.get<Budget>(`/users/${userId}/budgets/${budgetId}`);
    return response.data;
  },

  async createBudget(userId: number, payload: BudgetPayload) {
    const response = await api.post<Budget>(`/users/${userId}/budgets`, payload);
    return response.data;
  },

  async updateBudget(userId: number, budgetId: number, payload: BudgetPayload) {
    const response = await api.put<Budget>(`/users/${userId}/budgets/${budgetId}`, payload);
    return response.data;
  },

  async deleteBudget(userId: number, budgetId: number) {
    await api.delete(`/users/${userId}/budgets/${budgetId}`);
    return budgetId;
  }
};
